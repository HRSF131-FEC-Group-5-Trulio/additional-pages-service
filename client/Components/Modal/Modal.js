import React from "react";
import ReactDOM from 'react-dom';
import styled, {css} from 'styled-components';
import SearchBar from './SearchBar';
import ModalItems from './ModalItems/ModalItems.js';

const StyledModal = styled.div`
&.modal {
  position: fixed;
  top: 0;
  left: 0;
  width:100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.6);

  & .modal-main {
    position:fixed;
    width: 45%;
    height: 80%;
    top:50%;
    left:50%;
    transform: translate(-50%,-50%);
    border-radius: 20px;
    border: solid rgb(0, 120, 130);
    //border-color: green;
    background:rgb(224, 247, 248);
    //background: rgb(204, 255, 204);
  }
  &.display-block {
    display: flex;
    justify-content: center;
    text-align:center;

  }
  &.display-none {
    display: none;
  }
}
`
const sharedStyle = css`
position: absolute;
content: '';
height: 25px;
width: 2px;
background-color: #333;
float: left;
`
const CloseButton = styled.a`
&.close {
  position: absolute;
  left: 0%;
  top: 0%;
  float: left;
  width: 5%;
  height: 5%;
  padding-top: 10px;
  padding-left: 5px;
}
&.close:hover {
  opacity: 1;
}
&.close:before {
  ${sharedStyle}
}
&.close:after {
  ${sharedStyle}
}
&.close:before {
  transform: rotate(45deg);
}
&.close:after {
  transform: rotate(-45deg);
}
`

//Add a shadow??
class Modal extends React.Component {
  constructor(props) {
    super(props);

  }
//   getHighlightedText(text, highlight) {
//     // Split text on highlight term, include term itself into parts, ignore case
//     const parts = text.split(new RegExp(`(${highlight})`, 'gi'));
//     return <span>{parts.map(part => part.toLowerCase() === highlight.toLowerCase() ? <b>{part}</b> : part)}</span>;
// }
  stopPropagation(e) {
    e.stopPropagation();
  }

  render(){
    const showHideClassName = this.props.show ? 'modal display-block' : 'modal display-none';
    const {
      handleClose,
      favorites,
      children,
    } = this.props;
    return (
      <StyledModal className={showHideClassName} onClick={handleClose}>
        <section className='modal-main'  onClick={this.stopPropagation}>
        <CloseButton href="#" className="close" onClick={handleClose}/>
          {children}
          <ModalItems favorites={favorites} />
        </section>
      </StyledModal>
    );
  }
};

export default Modal