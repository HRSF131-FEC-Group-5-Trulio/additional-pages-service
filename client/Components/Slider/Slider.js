import React from "react";
import ReactDOM from 'react-dom';
import styled, {css} from 'styled-components';
import Listing from './Slides/Listing/Listing';
import LastSlide from './Slides/LastSlide/LastSlide';
import Buttons from './Buttons/Buttons';
const FlexContainer = styled.div`
&{
   display: flex;
    overflow-x: auto;
    scroll-behavior: smooth;
    right: 100px;
    margin-left: -8px;
    margin-right: -8px;
    border: solid;
    border-color: transparent;
 }
 &::-webkit-scrollbar {
    display: none;
  }
`;
const ContentSlider = styled.div`
position: relative;
box-sizing: border-box;
display: block;
outline: none;
margin-left: -8px;
 margin-right: -8px;
&::-webkit-scrollbar {
  display: none;
}
`
const NextAndPrevious = css`
  border: 1px solid rgb(232, 233, 234);
  position: absolute;
   width: 30px;
   height: 30px;
  background: #fff;
  color: black;
  top: 135px;
  border-radius: 50%;
  text-align: center;
  cursor: pointer;
  outline: none;
  line-height: 30px;
  text-align: center;

  &:hover {
    transform: scale(1.01);
    box-shadow: -1px 8px 21px -11px rgba(0,0,0,0.58);
  }

`
const PreviousButton = styled.i`
  ${NextAndPrevious}
  left: 0px;
`

const NextButton = styled.i`
  ${NextAndPrevious}
  right: 0px;
`

const Slider = ({handleScroll, showSlides, handleHeartClick, arrowButtonHandler,displayLeftArrow, displayRightArrow, properties}) => {
  return (
    <ContentSlider onScroll={handleScroll}>
      <FlexContainer>
      {showSlides.length > 0 ? showSlides.map((image, index) => (
        <Listing image={image} handleHeartClick= {handleHeartClick} index={index}/>
      )) : <div>{''}</div>}
      {<LastSlide propertiesLength={properties.length}/>}
        <PreviousArrowButton arrowButtonHandler={arrowButtonHandler} displayLeftArrow={displayLeftArrow} />
        <NextArrowButton arrowButtonHandler={arrowButtonHandler} displayRightArrow={displayRightArrow} />
      </FlexContainer>
    </ContentSlider>
  )
}

export default Slider;