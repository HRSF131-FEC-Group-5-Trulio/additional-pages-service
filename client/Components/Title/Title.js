import ReactDOM from 'react-dom';
import React from 'react';
import styled, {css} from 'styled-components';
import ShowFavorites from './ShowFavorites/ShowFavorites';

const TitleContainer = styled.div`
  display:flex;
  position: relative;
  width: 100%;
`

const LastSlide = ({showModal}) => (
    <TitleContainer>
      <h2 >Similar Homes You May Like</h2>
      <ShowFavorites showModal={showModal} />
    </TitleContainer>
  )

export default LastSlide