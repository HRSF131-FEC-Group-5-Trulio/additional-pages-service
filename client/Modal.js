import React from "react";
import ReactDOM from 'react-dom';
import styled, {css} from 'styled-components';
import SearchBar from './SearchBar';
import ModalItem from './ModalItem.js';

const StyledModal = styled.div`
&.modal {
  position: fixed;
  top: 0;
  left: 0;
  width:100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.6);

  & .modal-main {
    position:fixed;
    width: 45%;
    height: 80%;
    top:50%;
    left:50%;
    transform: translate(-50%,-50%);
    border-radius: 20px;
    border: solid;
    border-color: green;
    background: rgb(204, 255, 204);
  }
  &.display-block {
    display: flex;
    justify-content: center;
    text-align:center;

  }
  &.display-none {
    display: none;
  }
}
`
const sharedStyle = css`
position: absolute;
content: '';
height: 33px;
width: 2px;
background-color: #333;
float: left;
`
const CloseButton = styled.a`
&.close {
  position: absolute;
  left: 0%;
  top: 0%;
  float: left;
  width: 5%;
  height: 5%;
  padding-top: 5px;
  padding-left: 20px;
}
&.close:hover {
  opacity: 1;
}
&.close:before {
  ${sharedStyle}
}
&.close:after {
  ${sharedStyle}
}
&.close:before {
  transform: rotate(45deg);
}
&.close:after {
  transform: rotate(-45deg);
}
`
const FavoritesContainer = styled.div`
  text-align: left;
  top:50%;
   border: solid green;
   border-radius: 8px;
   height: 85%;
   width: 70%;
   margin-top: 20px;
   margin-left: auto;
   margin-right: auto;
   background: white;
   overflow-y: auto;
`

//Add a shadow??
class Modal extends React.Component {
  constructor(props) {
    super(props);

  }
  numberWithCommas(x, roundToNearest) {
    x = Math.round(x/roundToNearest)*roundToNearest
    return x.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",");
  }
  getHighlightedText(text, highlight) {
    // Split text on highlight term, include term itself into parts, ignore case
    const parts = text.split(new RegExp(`(${highlight})`, 'gi'));
    return <span>{parts.map(part => part.toLowerCase() === highlight.toLowerCase() ? <b>{part}</b> : part)}</span>;
}
  stopPropagation(e) {
    e.stopPropagation();
  }

  render(){
    const showHideClassName = this.props.show ? 'modal display-block' : 'modal display-none';
    return (
      <StyledModal className={showHideClassName} onClick={this.props.handleClose}>
        <section className='modal-main'  onClick={this.stopPropagation}>
        <CloseButton href="#" className="close" onClick={this.props.handleClose}/>
          {this.props.children}
          <FavoritesContainer>
            {this.props.favorites.length === 0? 'No items selected.' : this.props.favorites.map(favorite=> {return (
              <ModalItem favorite={favorite} numberWithCommas={this.numberWithCommas.bind()} />
            )}
            )}
          </FavoritesContainer>
        </section>
      </StyledModal>
    );
  }
};

export default Modal