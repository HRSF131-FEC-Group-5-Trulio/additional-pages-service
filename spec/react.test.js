import React from 'react'
import {mount, shallow} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import App from '../client/app.js';
import Modal from '../client/Components/Modal/Modal';
import Slider from '../client/Components/Slider/Slider';
import Title from '../client/Components/Title/Title';

import TestRenderer from 'react-test-renderer';
// import Link from '../Link.react';

test('shallow renders App component', () => {
  const wrapper = shallow(<div><App /></div>);
  expect(wrapper.find(App)).toHaveLength(1);
})

test('shallow renders Modal component', () => {
  const wrapper = shallow(<div><Modal /></div>);
  expect(wrapper.find(Modal)).toHaveLength(1);
})
test('shallow renders Slider component', () => {
  const wrapper = shallow(<div><Slider /></div>);
  expect(wrapper.find(Slider)).toHaveLength(1);
})




test('App component contains Title component', () => {
  const testRenderer = TestRenderer.create(<App />);
const testInstance = testRenderer.root;

expect(typeof testInstance.findByType(Title).props.showModal).toBe('function');
 //expect(testInstance.findByProps({show:false}).children).toHaveLength(2);
});
// describe('App debug mode', () => {
//   it('should render correctly in "debug" mode', () => {
//     const component = shallow(<App debug />);
//     expect(component).toMatchSnapshot();
//   });
// });

// import renderer from 'react-test-renderer';
// import Link from '../Link.react';

test('Title Render matches snapshot', () => {
  const showModal=() =>{
    this.setState({ show: true });
  }
  const component = TestRenderer.create(
    <Title showModal={showModal}/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // // manually trigger the callback
  // tree.props.onMouseEnter();
  // // re-rendering
  // tree = component.toJSON();
  // expect(tree).toMatchSnapshot();

  // // manually trigger the callback
  // tree.props.onMouseLeave();
  // // re-rendering
  // tree = component.toJSON();
  // expect(tree).toMatchSnapshot();
});