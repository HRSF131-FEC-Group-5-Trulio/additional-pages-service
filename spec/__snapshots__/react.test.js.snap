// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App debug mode should render correctly in "debug" mode 1`] = `
<app__OuterContainer>
  <LastSlide
    showModal={[Function]}
  />
  <Slider
    arrowButtonHandler={[Function]}
    displayLeftArrow={false}
    displayRightArrow={true}
    handleHeartClick={[Function]}
    handleScroll={[Function]}
    properties={Array []}
    showSlides={Array []}
  />
  <Modal
    favorites={Array []}
    handleClose={[Function]}
    show={false}
  >
    <app__ModalTitleContainer>
      <SearchBar
        favoriteList={Array []}
        onChangeHandler={[Function]}
      />
      <app__SortButton
        onClick={[Function]}
      >
        Sort By Price (
        &lt;
        )
      </app__SortButton>
    </app__ModalTitleContainer>
  </Modal>
</app__OuterContainer>
`;

exports[`Title Render matches snapshot 1`] = `
<div
  className="Title__TitleContainer-sc-1w9co3c-0 fspMKj"
>
  <h2>
    Similar Homes You May Like
  </h2>
  <button
    className="ShowFavorites__FavoritesLink-sc-1xsezjx-0 WpdnV"
    onClick={[Function]}
  >
    View your favorites list!
  </button>
</div>
`;
